marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -49,19 +51,31 @@ function getTypesInfo() {
 }
 
 function getArtifacts(artifactType) {
-    const artifacts = [];
     const queriesForAnd = [];
     const artifactLibrary =  getArtifactTypeLibrary(artifactType);
     for (const coll of artifactLibrary.getCollections()) {
         queriesForAnd.push(cts.collectionQuery(coll));
     }
     if (queriesForAnd.length) {
-        const results = cts.search(queriesForAnd.length > 1 ? cts.andQuery(queriesForAnd) : queriesForAnd[0]);
-        for (const result of results) {
-            artifacts.push(result.toObject());
+        if (artifactType == "loadData") {
+            return cts.search(queriesForAnd.length > 1 ? cts.andQuery(queriesForAnd) : queriesForAnd[0]).toArray();
+        } else {
+            return getArtifactsGroupByEntity(queriesForAnd)
         }
     }
-    return artifacts;
+    return [];
+}
+
+function getArtifactsGroupByEntity(queriesForAnd) {
```

Comment:
Is this returning artifacts, or configs, or settings? Are "configs" the same as "settings"? I don't think this is returning artifacts, which are the actual flows/stepDefinitions/mappings/etc. The variables indicate that these are "configs", so I'm wondering if those are the same as "settings".

Reply:
`[
  {
    "name": "CustomerRecords", 
    "config": [
      {
        "name": "Map 1",
        "description": "",
        "sourceFormat": "json",
        "selectedSource":"collection", //(collection  | query)
        "sourceQuery": "",
        "collections": ['Provider'],
        "lastUpdated": "2020-02-15T11:25:35.765299-08:00"
      },
      {
        "name": "Customer Mapping - New",
        "description": "",
        "sourceFormat": "xml",
        "selectedSource":"collection", //(collection  | query)
        "sourceQuery": "cts.uris(null,null,cts.andQuery[cts.CollectionQuery('Customer'),cts.elementAttributeValueQuery()])",
        "collections": ['Customer'],
        "lastUpdated": "2020-02-11T11:25:35.765299-08:00"
      }
    ]
  },
  {
    "name": "StudentCollegeRecordsNew",
    "config": [
      {
        "name": "Map 4",
        "description": "",
        "sourceFormat": "json",
        "selectedSource":"collection", //(collection  | query)
        "sourceQuery": "",
        "collections": ['Provider'],
        "lastUpdated": "2020-02-15T11:25:35.765299-08:00"
      },
      {
        "name": "StudentRec Mapping - New",
        "description": "",
        "sourceFormat": "xml",
        "selectedSource":collection", //(collection  | query),
        "sourceQuery": "",
        "collections": ['Customer'],
        "lastUpdated": "2020-02-11T11:25:35.765299-08:00",
      }
    ]
  }`

Reply:
I guess we had better use "configs" rather than "config" in the json above.

Reply:
So the use case is "Get me all the artifacts for each known entity type", right? 

If so, shouldn't "configs" be replaced with "artifacts"? You're then returning an array of objects, where each object has an entity type (I recommend replacing "name" with "entityType") and an array of artifacts that reference the entity type.

Reply:
OK, it is fine.